rundeckapp/src/main/groovy/com/dtolabs/rundeck/server/plugins/logging/SimpleDataFilterPlugin.groovy

```diff
@@ -106,6 +105,32 @@ See the [Java Pattern](https://docs.oracle.com/javase/8/docs/api/java/util/regex
     String invalidKeyPattern
 
     static class RegexValidator implements PropertyValidator {
+
+        @Override
+        boolean isValid(String value, Map<String,Object> props) throws ValidationException {
+            try {
+
+                def invalidKeyPattern = "";
+
+                if(props.containsKey("invalidKeyPattern")){
+                    invalidKeyPattern = props.get("invalidKeyPattern")
+                }
+
+                def compile = Pattern.compile(value)
+                Matcher m = compile.matcher("");
+
+                if(value != invalidKeyPattern && m.groupCount() == 0){
```

Comment:
this is potentially flaky, it would be better to have two separate validators for `invalidKeyPattern`, and the `regex` properties.  The invalidKeyPattern just needs a simple Regex validator, while the other has to check the number of groups in the regex. could subclass the simple Regex validator for the second case.